metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -739,8 +741,12 @@ def _process_schema(
             )
 
             for table in db_tables[dataset_name]:
-                table_columns = columns.get(table.name, []) if columns else []
+                identifier = BigqueryTableIdentifier(
+                    project_id, dataset_name, table.name
+                )
+                self.table_refs.add(str(BigQueryTableRef(identifier)))
```

Comment:
I acknowledge that it's kinda awkward to store refs like this, rather than a more "normal" value like the table identifier or something. I think this whole area is in need of a refactor though, so I hope this is okay for now.

Reply:
This does not seem to be right place to add identifier in table_refs. We should add it only after confirming this table fits into all other filters (such as allow deny pattern). I moved this inside `_process_table`.